File: ios/YiLuAn/Core/Models/FollowupReminder.swift
import Foundation

/// [F-07] 复诊提醒。
/// backend：`/orders/{id}/followup-reminders`、`/orders/me/followup-reminders`
struct FollowupReminder: Codable, Identifiable, Hashable {
    let id: String
    let userId: String?
    let orderId: String
    let remindAt: Date
    let status: String  // pending | sent | cancelled | failed
    let attempts: Int
    let note: String?
    let sentAt: Date?
    let createdAt: Date?
}

struct FollowupReminderListResponse: Decodable {
    let items: [FollowupReminder]
    let total: Int
}

struct CreateFollowupReminderRequest: Encodable {
    let orderId: String
    let remindAt: Date
    let note: String?
}

extension FollowupReminder {
    var statusLabel: String {
        let loc = LocalizationManager.shared
        switch status {
        case "pending":   return loc.t("followupReminders.statusPending")
        case "sent":      return loc.t("followupReminders.statusSent")
        case "cancelled": return loc.t("followupReminders.statusCancelled")
        case "failed":    return loc.t("followupReminders.statusFailed")
        default:          return status
        }
    }

    var canCancel: Bool { status == "pending" }
}

File: backend/app/schemas/followup_reminder.py
"""F-07 schemas — 复诊提醒."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class CreateFollowupReminderRequest(BaseModel):
    order_id: UUID = Field(..., description="订单 ID；必须是 completed / reviewed")
    remind_at: datetime = Field(..., description="提醒时间（UTC ISO 8601）")
    note: str | None = Field(None, max_length=140, description="自定义备注")


class FollowupReminderResponse(BaseModel):
    id: UUID
    user_id: UUID
    order_id: UUID
    remind_at: datetime
    status: str
    attempts: int
    note: str | None
    sent_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class FollowupReminderListResponse(BaseModel):
    items: list[FollowupReminderResponse]
    total: int
